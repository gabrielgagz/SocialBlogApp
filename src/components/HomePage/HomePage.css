.container-posts {
    padding: 0 4em 0 4em;
}

.HomePage__btn-more {
    color: white;
    background-color: #292b2c;
    font-weight: bold;
    font-size: smaller;
}

.HomePage__btn-more:hover {
    color: white;
    background-color: #292b2c;
}

@media only screen and (max-width: 600px) {

    .container-posts{
        padding: 0 0.5em 0 0.5em;
    }

}