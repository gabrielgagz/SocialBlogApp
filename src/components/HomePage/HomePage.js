import React, { useEffect, useState } from 'react';
import apiService from '../../services/apiService';
import { Post } from '../Post/Post';
import './HomePage.css';

export const HomePage = () => {

    const [ postState, setPostState] = useState({});

    useEffect(() => {
        
        (async () => {

            const data = await apiService('posts');
            setPostState( data );

        })();

    // eslint-disable-next-line react-hooks/exhaustive-deps
    }, []);

    return (
        <div className='container-fluid container-posts'>
            <div className='row row-cols-1 row-cols-md-3 my-5 justify-content-center'>
                { 
                    postState.length > 0 && postState.map( ( postData ) => {
                        return <Post data={ postData } key={ postData.id }/>
                    }) 
                }
            </div>
        </div>
    )
}
