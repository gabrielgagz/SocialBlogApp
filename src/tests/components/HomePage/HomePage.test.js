import { React } from 'react';
import { HomePage } from '../../../components/HomePage/HomePage';
import * as redux from 'react-redux';
import { shallow } from 'enzyme';

describe('<HomePage /> tests', () => {
    
    let spyOnUseSelector;
    let spyOnUseDispatch;
    let mockDispatch;

    beforeEach(() => {
        spyOnUseSelector = jest.spyOn(redux, 'useSelector');
        spyOnUseSelector.mockReturnValue([{ 
            postId: 1,
            id: 1,
            name: 'id labore ex et quam laborum',
            email: 'Eliseo@gardner.biz',
            body: 'laudantium enim quasi est quidem magnam voluptate ipsam eos\ntempora quo necessitatibus\ndolor quam autem quasi\nreiciendis et nam sapiente accusantium' 
        }]);

        spyOnUseDispatch = jest.spyOn(redux, 'useDispatch');
        mockDispatch = jest.fn();
        spyOnUseDispatch.mockReturnValue(mockDispatch);
    });

    afterEach(() => {
        jest.restoreAllMocks();
    });

    test('Component must properly render', () => {

        const wrapper = shallow(<HomePage />);
        expect(wrapper).toMatchSnapshot();

    });

});
