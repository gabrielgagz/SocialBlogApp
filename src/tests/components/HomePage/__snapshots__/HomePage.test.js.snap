// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HomePage /> tests Component must properly render 1`] = `
<div
  className="container-fluid"
>
  <div
    className="row row-cols-1 row-cols-md-3 my-5 justify-content-center"
  />
  <button
    className="btn HomePage__btn-more mt-3 mb-5 p-3 mx-auto d-block"
    onClick={[Function]}
  >
    VIEW MORE POSTS
  </button>
</div>
`;
